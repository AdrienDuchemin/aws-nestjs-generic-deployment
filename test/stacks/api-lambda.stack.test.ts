import { expect as expectCDK, SynthUtils, countResources } from '@aws-cdk/assert'
import { Code } from '@aws-cdk/aws-lambda'
import { App } from '@aws-cdk/core'
import { ApiLambdaStack, IApiLambdaStackProps } from '../../src/stacks/api-lambda.stack'

describe('ApiLambdaStack', () => {
  let stack: ApiLambdaStack

  beforeAll(()=> {
    const props: IApiLambdaStackProps = {
      lambdaProps: {
        code: Code.fromInline('lambda'),
        handler: 'handler',
      }
    }
    stack = new ApiLambdaStack(new App(), 'ApiLambdaStack', props)
  })

  it('should create one Lambda', () => {
    expectCDK(stack).to(countResources("AWS::Lambda::Function", 1))
  })

  it('should create one Api Gateway', () => {
    expectCDK(stack).to(countResources("AWS::ApiGateway::RestApi", 1))
  })

  it('should not have changed', () => {
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot()
  })
})