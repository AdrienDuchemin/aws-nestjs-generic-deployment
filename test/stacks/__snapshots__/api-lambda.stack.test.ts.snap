// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApiLambdaStack should not have changed 1`] = `
Object {
  "Outputs": Object {
    "ApiLambdaStackLambdaApiLambdaStackRestApiEndpointB1631A72": Object {
      "Value": Object {
        "Fn::Join": Array [
          "",
          Array [
            "https://",
            Object {
              "Ref": "ApiLambdaStackLambdaApiLambdaStackRestApi53EF09E4",
            },
            ".execute-api.",
            Object {
              "Ref": "AWS::Region",
            },
            ".",
            Object {
              "Ref": "AWS::URLSuffix",
            },
            "/",
            Object {
              "Ref": "ApiLambdaStackLambdaApiLambdaStackRestApiDeploymentStageprodB52A1F58",
            },
            "/",
          ],
        ],
      },
    },
  },
  "Resources": Object {
    "ApiLambdaStackLambda78623D88": Object {
      "DependsOn": Array [
        "ApiLambdaStackLambdaServiceRole29C5619E",
      ],
      "Properties": Object {
        "Code": Object {
          "ZipFile": "lambda",
        },
        "FunctionName": "ApiLambdaStackLambda",
        "Handler": "handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "ApiLambdaStackLambdaServiceRole29C5619E",
            "Arn",
          ],
        },
        "Runtime": "nodejs12.x",
        "Timeout": 28,
      },
      "Type": "AWS::Lambda::Function",
    },
    "ApiLambdaStackLambdaAllowEventRuleApiLambdaStackApiLambdaStackLambdaApiLambdaStackRule61C0AE32478FEA25": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "ApiLambdaStackLambda78623D88",
            "Arn",
          ],
        },
        "Principal": "events.amazonaws.com",
        "SourceArn": Object {
          "Fn::GetAtt": Array [
            "ApiLambdaStackLambdaApiLambdaStackRuleD2D84EB7",
            "Arn",
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "ApiLambdaStackLambdaApiLambdaStackRestApi53EF09E4": Object {
      "Properties": Object {
        "Name": "ApiLambdaStackRestApi",
      },
      "Type": "AWS::ApiGateway::RestApi",
    },
    "ApiLambdaStackLambdaApiLambdaStackRestApiANY4B6E1775": Object {
      "Properties": Object {
        "AuthorizationType": "NONE",
        "HttpMethod": "ANY",
        "Integration": Object {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                Object {
                  "Ref": "AWS::Region",
                },
                ":lambda:path/2015-03-31/functions/",
                Object {
                  "Fn::GetAtt": Array [
                    "ApiLambdaStackLambda78623D88",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": Object {
          "Fn::GetAtt": Array [
            "ApiLambdaStackLambdaApiLambdaStackRestApi53EF09E4",
            "RootResourceId",
          ],
        },
        "RestApiId": Object {
          "Ref": "ApiLambdaStackLambdaApiLambdaStackRestApi53EF09E4",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "ApiLambdaStackLambdaApiLambdaStackRestApiANYApiPermissionApiLambdaStackApiLambdaStackLambdaApiLambdaStackRestApiA3584E23ANY22DEDB35": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "ApiLambdaStackLambda78623D88",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              Object {
                "Ref": "AWS::Region",
              },
              ":",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "ApiLambdaStackLambdaApiLambdaStackRestApi53EF09E4",
              },
              "/",
              Object {
                "Ref": "ApiLambdaStackLambdaApiLambdaStackRestApiDeploymentStageprodB52A1F58",
              },
              "/*/",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "ApiLambdaStackLambdaApiLambdaStackRestApiANYApiPermissionTestApiLambdaStackApiLambdaStackLambdaApiLambdaStackRestApiA3584E23ANY8A5F55BA": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "ApiLambdaStackLambda78623D88",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              Object {
                "Ref": "AWS::Region",
              },
              ":",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "ApiLambdaStackLambdaApiLambdaStackRestApi53EF09E4",
              },
              "/test-invoke-stage/*/",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "ApiLambdaStackLambdaApiLambdaStackRestApiAccountF0DDE2B7": Object {
      "DependsOn": Array [
        "ApiLambdaStackLambdaApiLambdaStackRestApi53EF09E4",
      ],
      "Properties": Object {
        "CloudWatchRoleArn": Object {
          "Fn::GetAtt": Array [
            "ApiLambdaStackLambdaApiLambdaStackRestApiCloudWatchRoleB3344189",
            "Arn",
          ],
        },
      },
      "Type": "AWS::ApiGateway::Account",
    },
    "ApiLambdaStackLambdaApiLambdaStackRestApiCloudWatchRoleB3344189": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "apigateway.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AmazonAPIGatewayPushToCloudWatchLogs",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "ApiLambdaStackLambdaApiLambdaStackRestApiDeploymentF545FAEDbd212b8b04be689a445f877db702b0d3": Object {
      "DependsOn": Array [
        "ApiLambdaStackLambdaApiLambdaStackRestApiproxyANY3330E380",
        "ApiLambdaStackLambdaApiLambdaStackRestApiproxyE132521F",
        "ApiLambdaStackLambdaApiLambdaStackRestApiANY4B6E1775",
      ],
      "Properties": Object {
        "Description": "Automatically created by the RestApi construct",
        "RestApiId": Object {
          "Ref": "ApiLambdaStackLambdaApiLambdaStackRestApi53EF09E4",
        },
      },
      "Type": "AWS::ApiGateway::Deployment",
    },
    "ApiLambdaStackLambdaApiLambdaStackRestApiDeploymentStageprodB52A1F58": Object {
      "Properties": Object {
        "DeploymentId": Object {
          "Ref": "ApiLambdaStackLambdaApiLambdaStackRestApiDeploymentF545FAEDbd212b8b04be689a445f877db702b0d3",
        },
        "RestApiId": Object {
          "Ref": "ApiLambdaStackLambdaApiLambdaStackRestApi53EF09E4",
        },
        "StageName": "prod",
      },
      "Type": "AWS::ApiGateway::Stage",
    },
    "ApiLambdaStackLambdaApiLambdaStackRestApiproxyANY3330E380": Object {
      "Properties": Object {
        "AuthorizationType": "NONE",
        "HttpMethod": "ANY",
        "Integration": Object {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                Object {
                  "Ref": "AWS::Region",
                },
                ":lambda:path/2015-03-31/functions/",
                Object {
                  "Fn::GetAtt": Array [
                    "ApiLambdaStackLambda78623D88",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": Object {
          "Ref": "ApiLambdaStackLambdaApiLambdaStackRestApiproxyE132521F",
        },
        "RestApiId": Object {
          "Ref": "ApiLambdaStackLambdaApiLambdaStackRestApi53EF09E4",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "ApiLambdaStackLambdaApiLambdaStackRestApiproxyANYApiPermissionApiLambdaStackApiLambdaStackLambdaApiLambdaStackRestApiA3584E23ANYproxy6F74C4BC": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "ApiLambdaStackLambda78623D88",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              Object {
                "Ref": "AWS::Region",
              },
              ":",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "ApiLambdaStackLambdaApiLambdaStackRestApi53EF09E4",
              },
              "/",
              Object {
                "Ref": "ApiLambdaStackLambdaApiLambdaStackRestApiDeploymentStageprodB52A1F58",
              },
              "/*/{proxy+}",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "ApiLambdaStackLambdaApiLambdaStackRestApiproxyANYApiPermissionTestApiLambdaStackApiLambdaStackLambdaApiLambdaStackRestApiA3584E23ANYproxyE4822CC7": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "ApiLambdaStackLambda78623D88",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              Object {
                "Ref": "AWS::Region",
              },
              ":",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "ApiLambdaStackLambdaApiLambdaStackRestApi53EF09E4",
              },
              "/test-invoke-stage/*/{proxy+}",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "ApiLambdaStackLambdaApiLambdaStackRestApiproxyE132521F": Object {
      "Properties": Object {
        "ParentId": Object {
          "Fn::GetAtt": Array [
            "ApiLambdaStackLambdaApiLambdaStackRestApi53EF09E4",
            "RootResourceId",
          ],
        },
        "PathPart": "{proxy+}",
        "RestApiId": Object {
          "Ref": "ApiLambdaStackLambdaApiLambdaStackRestApi53EF09E4",
        },
      },
      "Type": "AWS::ApiGateway::Resource",
    },
    "ApiLambdaStackLambdaApiLambdaStackRuleD2D84EB7": Object {
      "Properties": Object {
        "ScheduleExpression": "rate(5 minutes)",
        "State": "ENABLED",
        "Targets": Array [
          Object {
            "Arn": Object {
              "Fn::GetAtt": Array [
                "ApiLambdaStackLambda78623D88",
                "Arn",
              ],
            },
            "Id": "Target0",
            "Input": "{\\"ping\\":true}",
          },
        ],
      },
      "Type": "AWS::Events::Rule",
    },
    "ApiLambdaStackLambdaServiceRole29C5619E": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
  },
}
`;
